evalml/tests/pipeline_tests/test_component_graph.py

```diff
@@ -2024,3 +2024,35 @@ def test_component_graph_repr():
     expected_repr = "{'Imputer': ['Imputer', 'X', 'y'], 'OHE': ['One Hot Encoder', 'Imputer.x', 'y'], 'Random Forest Classifier': ['Random Forest Classifier', 'OHE.x', 'y']}"
     component_graph = ComponentGraph(component_dict_with_objs)
     assert repr(component_graph) == expected_repr
+
+
+@patch("evalml.pipelines.components.estimators.LogisticRegressionClassifier.fit")
+@pytest.mark.parametrize("sampler", ["Undersampler", "SMOTE Oversampler"])
+def test_component_graph_compute_final_component_features_with_sampler(
```

Comment:
Original issue was in using a sampler while calculating permutation importance. However, the issue wasn't with permutation importance specifically but rather, the pipeline calling `compute_estimator_features` which called `compute_final_component_features`. Hence, testing this here :)